### PR TITLE
[plugin-mobile-app] Handle the exception if status bar height from session details is null

### PR DESCRIPTION
**Pull request by EDbarvinsky**
Received exception in [CI example](https://github.com/vividus-framework/vividus/runs/4689829224?check_suite_focus=true#step:21:2124)

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java [hunk 1]
```diff
@@ -134,6 +134,10 @@ private float calculateDpr()
     private int getStatBarHeight()
     {
         Long statBarHeight = getSessionDetail("statBarHeight");
+        if (statBarHeight == null)
+        {
+            throw new IllegalArgumentException("Unable to receive status bar height. Received value is null.");
+        }
         return statBarHeight.intValue();
     }
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java:142]:
```suggestion
    private int getStatBarHeightSafely()
    {
        Long statBarHeight = getStatBarHeightUnsafely();
        if (statBarHeight == null)
        {
            throw new IllegalArgumentException("Unable to receive status bar height. Received value is null.");
        }
        return statBarHeight.intValue();
    }
    
    private Long getStatBarHeightUnsafely()
    {
        return (Long) getSessionDetail("statBarHeight");
    }
```

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public int getStatusBarSize()
             {
                 return getStatBarHeight();
             }
-            catch (WebDriverException e)
+            catch (WebDriverException | IllegalArgumentException e)
             {
                 // Handling SauceLabs error:
                 // org.openqa.selenium.WebDriverException:
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java:82]:
```java
            Long statBarHeight;
            try
            {
                statBarHeight = getStatBarHeightUnsafely();
            }
            catch (WebDriverException)
            {
                statBarHeight = null;
            }
            // Handling SauceLabs error:
            ...
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java [hunk 3]
```diff
@@ -72,21 +72,27 @@ public int getStatusBarSize()
...
 
                 // Appium 1.21.0 or higher is required
+            }
+            if (null == statBarHeight)
+            {
                 Map<String, ?> deviceScreenInfo = javascriptActions.executeScript("mobile:deviceScreenInfo");
                 Map<String, ?> statusBarSize = (Map<String, ?>) deviceScreenInfo.get("statusBarSize");
                 return ((Number) statusBarSize.get(HEIGHT)).intValue();
             }
+            return statBarHeight.intValue();
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java:95]:
```java
Number statBarHeight;
try
{
    statBarHeight = getStatBarHeightUnsafely();
}
catch (WebDriverException e)
{
    // Handling SauceLabs error:
    // org.openqa.selenium.WebDriverException:
    // failed serving request GET https://production-sushiboat.default/wd/hub/session/XXXX
    statBarHeight = null;
}
if (null == statBarHeight)
{
    // Appium 1.21.0 or higher is required
    Map<String, ?> deviceScreenInfo = javascriptActions.executeScript("mobile:deviceScreenInfo");
    Map<String, ?> statusBarSize = (Map<String, ?>) deviceScreenInfo.get("statusBarSize");
    statBarHeight = (Number) statusBarSize.get(HEIGHT));
}
return statBarHeight.intValue();
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java [hunk 4]
```diff
@@ -131,9 +136,18 @@ private float calculateDpr()
         }
     }
 
-    private int getStatBarHeight()
+    private int getStatBarHeightSafely()
     {
-        Long statBarHeight = getSessionDetail("statBarHeight");
+        Long statBarHeight = getStatBarHeightUnsafely();
+        if (statBarHeight == null)
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java:142]: we usually use Validate.isTrue from apache commons lang3 lib

**Review by uarlouski — APPROVED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java [hunk 5]
```diff
@@ -131,9 +136,18 @@ private float calculateDpr()
...
+        {
+            throw new IllegalArgumentException("Unable to receive status bar height. Received value is null.");
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java:144]:
```suggestion
            throw new IllegalStateException("Unable to receive status bar height. Received value is null.");
```

File: vividus-plugin-mobile-app/src/test/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManagerTests.java [hunk 6]
```diff
@@ -121,13 +125,40 @@ void shouldProviderStatusBarHeightForIosWhenSauceLabsThrowAnError()
...
+        mockCapabilities(MobilePlatform.ANDROID);
+        when(webDriverProvider.getUnwrapped(HasAndroidDeviceDetails.class)).thenThrow(new WebDriverException("Ooops!"));
+        ExecutesMethod executingMethodDriver = mock(ExecutesMethod.class);
+
+        when(webDriverProvider.getUnwrapped(ExecutesMethod.class)).thenReturn(executingMethodDriver);
+        Response response = new Response();
+        response.setValue(Collections.EMPTY_MAP);
+
+        when(executingMethodDriver.execute(GET_SESSION_COMMAND)).thenReturn(response);
+        Exception exception = assertThrows(IllegalArgumentException.class, driverManager::getStatusBarSize);
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/test/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManagerTests.java:158]:
```suggestion
        var exception = assertThrows(IllegalStateException.class, driverManager::getStatusBarSize);
```

File: vividus-plugin-mobile-app/src/test/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManagerTests.java [hunk 7]
```diff
@@ -121,13 +125,40 @@ void shouldProviderStatusBarHeightForIosWhenSauceLabsThrowAnError()
...
+        when(webDriverProvider.getUnwrapped(HasAndroidDeviceDetails.class)).thenThrow(new WebDriverException("Ooops!"));
+        ExecutesMethod executingMethodDriver = mock(ExecutesMethod.class);
+
+        when(webDriverProvider.getUnwrapped(ExecutesMethod.class)).thenReturn(executingMethodDriver);
+        Response response = new Response();
+        response.setValue(Collections.EMPTY_MAP);
+
+        when(executingMethodDriver.execute(GET_SESSION_COMMAND)).thenReturn(response);
+        var exception = assertThrows(IllegalStateException.class, driverManager::getStatusBarSize);
+        assertEquals("Unable to receive status bar height. Received value is null.", exception.getMessage());
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/test/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManagerTests.java:159]:
```suggestion
        assertEquals("Unable to receive status bar height. Received value is null", exception.getMessage());
```

**Pull request merged**